.github/workflows/cmake.yml

```
@@ -94,45 +94,76 @@ jobs:
       - uses: actions/checkout@v4
         with:
           submodules: recursive
+      - uses: actions/checkout@v4
+        with:
+          path: apache-arrow
+          repository: apache/arrow
       - name: Prepare ccache
         run: |
           echo "CCACHE_DIR=${PWD}/ccache" >> ${GITHUB_ENV}
-      - name: Cache ccache
+      - name: Cache Groonga ccache
         uses: actions/cache@v4
         with:
           path: ccache
-          key: cmake-ubuntu-${{ matrix.cc }}-ccache-${{ hashFiles('lib/**', 'src/**', 'plugins/**', 'include/**') }}
-          restore-keys: cmake-ubuntu-${{ matrix.cc }}-ccache-
-      - name: Enable Apache Arrow repository
+          key: cmake-ubuntu-groonga-${{ matrix.cc }}-ccache-${{ hashFiles('lib/**', 'src/**', 'plugins/**', 'include/**') }}
+          restore-keys: |
+            cmake-ubuntu-groonga-${{ matrix.cc }}-ccache-
+            cmake-ubuntu-groonga-ccache-
+      - name: Configure ccache for Apache Arrow
+        run: |
+          echo "CCACHE_DIR_ARROW=${PWD}/ccache-arrow" >> ${GITHUB_ENV}
+          echo "CMAKE_CXX_COMPILER_LAUNCHER_ARROW=ccache" >> ${GITHUB_ENV}
+      - name: Cache Apache Arrow ccache
+        uses: actions/cache@v4
+        with:
+          path: ccache-arrow
+          key: cmake-ubuntu-apache-arrow-${{ matrix.cc }}-ccache-${{ hashFiles('apache-arrow/cpp/src/**') }}
+          restore-keys: |
+            cmake-ubuntu-apache-arrow-${{ matrix.cc }}-ccache-
+            cmake-ubuntu-apache-arrow-ccache-
+      - name: Install dependencies for Apache Arrow
```

It's a package that is also used in Groonga builds, so maybe we can rename it or do `Install packages` here.

```suggestion
      - name: Install dependencies
```